agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1668,7 +1668,7 @@ bool backhaul_manager::handle_slave_backhaul_message(std::shared_ptr<SSlaveSocke
         LOG(DEBUG) << "ACTION_BACKHAUL_HOSTAP_VAPS_LIST_UPDATE_NOTIFICATION received from iface "
                    << soc->hostap_iface;
         if (m_agent_ucc_listener) {
-            auto msg = cmdu_rx.getClass<
+            auto msg = beerocks_header->addClass<
```

Comment:
This bug has slipped from my eyes, so I wonder if there is any way to fail while using the wrong API?

Reply:
We need to convert beerocks_header like I did the AttrList and make addClass protected so it can't be used from outside. This will uncover all bugs, but should be done in a separate PR.

Reply:
It won't uncover anything... The issue that we had here is that we're calling cmdu_rx.getClass<...> with a class that is not a TLV.

If we want to detect this, we would need to check that the class is really a 1905 TLV, using that trick that @Galvien added to verify that a template instantiation is of the correct class. Small showstopper for that: there is currently no base class for 1905 TLV...